hnn_core/network_builder.py

```diff
@@ -298,15 +291,21 @@ def _build(self):
 
         self._clear_last_network_objects()
 
-        self._gid_assign()
+        # initialise vectors for synaptic currents based on simulation params
+        n_times = np.arange(0., self.net.params['tstop'],
+                            self.net.params['dt']).size + 1
 
         # Create a h.Vector() with size 1xself.N_t, zero'd
         self.current = {
-            'L5Pyr_soma': h.Vector(self.net.n_times, 0),
-            'L2Pyr_soma': h.Vector(self.net.n_times, 0),
+            'L5Pyr_soma': h.Vector(n_times, 0),
+            'L2Pyr_soma': h.Vector(n_times, 0),
         }
 
         self._create_cells_and_feeds()
+
+        self._gid_assign()
```

Comment:
I have been thinking a bit about what is inside `_gid_assign`. I think ideally `NetworkBuilder` should take `rank` in the `__init__` that is passed from `mpi_child.py` and then `_clone_and_simulate`. Then inside the `__init__` you do:

```py
self.rank = rank
```

and use `self.rank` inside `_gid_assign` instead of `_get_rank`. This will make it clear that `NetworkBuilder` is actually on a particular node. That way you will also limit nested calls to Neuron which we know can behave somewhat arbitrarily. Probably for another PR though

Reply:
I like it, will keep in mind, as I'm pretty sure it'll belong to this PR.

Reply:
Wasn't hard, just added rank and n_hosts as attributes to `NetworkBuilder` objects.

Reply:
I take it back: something exploded so I rolled it back. Separate PR!